macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
 
 import AppKit
 
+// MARK: - Button Formatting Parameters
+
+/// Indicates the size of the button
+@objc(MSFButtonSize)
+public enum ButtonSize: Int, CaseIterable {
+	case large
+	case small
+}
+
 /// Indicates what style our button is drawn as
 @objc(MSFButtonStyle)
 public enum ButtonStyle: Int, CaseIterable {
-	case primaryFilled	// Solid fill color
-	case primaryOutline	// Clear fill color, solid outline
-	case borderless		// Clear fill color, clear outline
+	/// No defined colors; user to specify.
+	case none
+
+	/// Accent color fill, white text/image.
+	case primary
+
+	/// Light mode: white fill, black text/image, light gray outline.
+	/// Dark mode: dark gray fill, white text/image, subtle gray outline.
+	/// Pressed: accent color fill, white text/image, no outline (same as primary style).
+	case secondary
+
+	/// Light mode: light gray fill, black text/image.
+	/// Dark mode: dark gray fill, white text/image.
+	case acrylic
+
+	/// Accent color text/image, no fill or outline.
+	case borderless
+}
+
+/// Combination of all formatting parameters.
+public struct ButtonFormat {
+	var size: ButtonSize
+	var style: ButtonStyle
+	var accentColor: NSColor?
```

Comment:
Since we give this a default value that is not nil, can this be non-optional?

Reply:
Fixed.